crates/burn-train/src/lib.rs

```diff
@@ -25,3 +25,156 @@ pub use learner::*;
+        const N_SAMPLES: usize = 200;
+        const N_CLASSES: usize = 4;
+
+        let error_mask = {
+            let mut rng = &mut rand::thread_rng();
+            let change_idx = Tensor::from_floats(
+                (0..N_SAMPLES)
+                    .into_iter()
+                    .choose_multiple(
+                        &mut rng,
+                        (N_SAMPLES as f32 * ERROR_PER_SAMPLE_RATE) as usize,
+                    )
+                    .as_slice(),
+                device,
+            );
+            let values = change_idx.ones_like();
+            let mask =
+                Tensor::zeros(Shape::new([N_SAMPLES]), device).scatter(0, change_idx.int(), values);
+            mask.unsqueeze_dim(1).bool()
+        };
+
+        let (targets, changed_targets) = match classification_type {
+            ClassificationType::Binary => {
+                let targets = Tensor::<TestBackend, 2>::random(
+                    Shape::new([N_SAMPLES, 1]),
+                    Distribution::Bernoulli(TRUE_POSITIVE_RATE),
+                    device,
+                )
+                .bool();
+
+                let changed_targets = targets.clone().not_equal(error_mask);
+                (targets, changed_targets.float())
+            }
+            ClassificationType::Multiclass => {
+                let mut classes_changes =
+                    Tensor::<TestBackend, 2>::random([N_SAMPLES, 2], Distribution::Default, device)
+                        .mul_scalar(4)
+                        .int()
+                        .float()
+                        .chunk(2, 1);
+                let (classes, changes) = (classes_changes.remove(0), classes_changes.remove(0));
+                let changed_classes = (classes.clone()
+                    + changes.clamp(1, 2) * error_mask.clone().float())
+                    % N_CLASSES as f32;
+                (
+                    one_hot_encode(classes, N_CLASSES).bool(),
+                    one_hot_encode(changed_classes, N_CLASSES),
+                )
+            }
+            ClassificationType::Multilabel => {
+                let targets = Tensor::<TestBackend, 2>::random(
+                    [N_SAMPLES, N_CLASSES - 1],
+                    Distribution::Default,
+                    device,
+                )
+                .greater_elem(THRESHOLD);
+
+                (targets.clone(), (targets.float() + error_mask.float()) % 2)
+            }
+        };
+        let predictions = changed_targets
+            .random_like(Distribution::Uniform(0.0, THRESHOLD - 0.1))
+            .sub(changed_targets.clone())
+            .abs();
+
+        (
+            ClassificationInput::new(predictions, targets.clone()),
+            targets.float().sub(changed_targets),
+        )
+    }
+
+    use burn_core::tensor::cast::ToElement;
+    use burn_core::tensor::TensorData;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn test_predictions_targets_match() {
```

Comment:
By simplifying the test inputs we should be able to remove this test (whose sole purpose is to test the reused code).

Reply:
changed